chains/evm/contracts/test/DonIDClaimer.t.sol

```diff
@@ -0,0 +1,131 @@
+    assertEq(s_donIDClaimer.getNextDONId(), INITIAL_CLAIM_ID + 1, "Next DON ID should be incremented to 101");
+  }
+
+  function test_SyncNextDONIdWithOffset() public {
+    vm.expectEmit();
+    emit DonIDClaimer.DonIDSynced(INITIAL_CLAIM_ID + 10);
+
+    vm.prank(s_deployer);
+    s_donIDClaimer.syncNextDONIdWithOffset(10);
+    assertEq(s_donIDClaimer.getNextDONId(), INITIAL_CLAIM_ID + 10, "Next DON ID should be 110 after offset");
+  }
+
+  function test_SetAuthorizedDeployer() public {
+    vm.expectEmit();
+    emit DonIDClaimer.AuthorizedDeployerSet(s_unauthorized, true);
+
+    vm.prank(s_owner);
+    s_donIDClaimer.setAuthorizedDeployer(s_unauthorized, true);
+    assertTrue(s_donIDClaimer.isAuthorizedDeployer(s_unauthorized), "Address should be authorized");
+  }
+
+  function test_SetAuthorizedDeployerRevoked() public {
+    vm.expectEmit();
+    emit DonIDClaimer.AuthorizedDeployerSet(s_deployer, false);
+
+    vm.prank(s_owner);
+    s_donIDClaimer.setAuthorizedDeployer(s_deployer, false);
+    assertFalse(s_donIDClaimer.isAuthorizedDeployer(s_deployer), "Deployer should be deauthorized");
+  }
+
+  function test_ClaimMultipleDONIds() public {
+    vm.startPrank(s_owner);
+    s_donIDClaimer.setAuthorizedDeployer(s_deployer, true);
+    vm.stopPrank();
+
+    vm.startPrank(s_deployer);
+    for (uint32 i = 0; i < 5; i++) {
+      uint32 claimed = s_donIDClaimer.claimNextDONId();
+      assertEq(claimed, INITIAL_CLAIM_ID + i, "Should claim incrementally");
+    }
+    vm.stopPrank();
+
+    assertEq(s_donIDClaimer.getNextDONId(), INITIAL_CLAIM_ID + 5, "Next ID should be 5 ahead");
+  }
+
+  function test_RevokeThenReauthorizeDeployer() public {
+    vm.startPrank(s_owner);
+    s_donIDClaimer.setAuthorizedDeployer(s_deployer, false);
+    assertFalse(s_donIDClaimer.isAuthorizedDeployer(s_deployer), "Should be deauthorized");
+
+    s_donIDClaimer.setAuthorizedDeployer(s_deployer, true);
+    assertTrue(s_donIDClaimer.isAuthorizedDeployer(s_deployer), "Should be re-authorized");
+    vm.stopPrank();
+  }
+
+  // Reverts
+  function test_RevertWhen_UnauthorizedSenderClaimReverts() public {
+    vm.expectRevert(abi.encodeWithSelector(DonIDClaimer.AccessForbidden.selector, s_unauthorized));
+    vm.prank(s_unauthorized);
+    s_donIDClaimer.claimNextDONId();
+  }
+
+  function test_RevertWhen_UnauthorizedSetAuthorizedDeployer() public {
+    vm.expectRevert(Ownable2Step.OnlyCallableByOwner.selector);
+    vm.prank(s_unauthorized);
+    s_donIDClaimer.setAuthorizedDeployer(s_unauthorized, true);
+  }
+
+  function test_RevertWhen_ConstructorWithZeroAddress() public {
+    vm.expectRevert(abi.encodeWithSelector(DonIDClaimer.ZeroAddressNotAllowed.selector));
+    new DonIDClaimer(address(0));
+  }
+
+  function test_RevertWhen_UnauthorizedSyncNextDONId() public {
+    vm.expectRevert(abi.encodeWithSelector(DonIDClaimer.AccessForbidden.selector, s_unauthorized));
+    vm.prank(s_unauthorized);
```

Comment:
nit: no logic between expect and the call, prank should come first, please fix in the other tests as well

Reply:
this should have it: https://github.com/smartcontractkit/chainlink-ccip/pull/819/commits/509a863ffabbf64d5db27bf42e302db57afb64fe